internal/civo/sdk/client.go

```diff
@@ -0,0 +1,80 @@
+package sdk
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/konstructio/dropkick/internal/civo/sdk/json"
+)
+
+// Civoer is the interface that represents a high-level Civo client. It differs
+// from the JSONClient interface in that this is a higher-level abstraction that
+// is used to interact with the Civo API.
+type Civoer interface {
+	Do(ctx context.Context, location, method string, output interface{}, params map[string]string) error
+	GetRegion() string
+}
+
+// JSONClient is an interface that allows us to make requests to the Civo API.
+// It's used to convey the low-level SDK client that can use generics to make
+// requests to the Civo API.
+type JSONClient interface {
+	Do(ctx context.Context, location, method string, output interface{}, params map[string]string) error
+	GetClient() *http.Client
+	GetEndpoint() string
+}
+
+// Client is a Civo client.
+type Client struct {
+	region string
+
+	client    *http.Client
+	requester JSONClient
+}
+
+// Option is a functional option for the Client.
+type Option func(*Client) error
+
+// WithJSONClient is an option to set a custom JSON client.
+// The client can be nil, in which case http.DefaultClient will be used.
+// The endpoint is the base URL for the Civo API.
+// The bearerToken is the token to authenticate with the Civo API.
+func WithJSONClient(client *http.Client, endpoint, bearerToken string) Option {
+	return func(c *Client) error {
+		c.requester = json.New(client, endpoint, bearerToken)
+		return nil
+	}
+}
+
+// WithRegion is an option to set the region.
+func WithRegion(region string) Option {
+	return func(c *Client) error {
+		c.region = region
+		return nil
+	}
+}
+
+// New creates a new Civo client.
+func New(opts ...Option) (*Client, error) {
+	c := &Client{
+		client: http.DefaultClient,
```

Comment:
Should we use a custom defaultClient with at least some timeout configuration instead of `http.DefaultClient`?

Reply:
Not a bad idea! There was an option to specify a custom client but I wasn't calling it anywhere. Might use the one that has debugging enabled.

Reply:
Yep, that makes sense. I mean, I know that in the end, we are the ones using and setting up this. But, you know, having some minimal config helps avoid trusting the other side of the pipe too much!

Reply:
Absolutely, let me double check. Last time I checked the "get load balancers" endpoint from Civo seems to be the slowest to load.

I'll use that as a baseline to define some timeouts.

Reply:
Done!